Library/Formula/rust.rb

```diff
@@ -15,6 +37,14 @@ class Rust < Formula
     sha256 "5194f3ae7512d842ac8085628c31b1c36ceae0fb15382f3a1982bc9e80609ae3" => :mountain_lion
   end
 
+  option "with-brewed-llvm", "Use Homebrew's LLVM to build rust (less optimized, more fixes)"
+  option "with-brewed-rust", "Use Homebrew's rust to build cargo"
```

Comment:
Just to sum up things and be clear about what we want:

`"with-brewed-rust"`: Unfortunately this won't build currently on stable (i.e beta) due to feature gates. I added it so we can easily test if it works, and as soon as it works ([probably next beta](https://github.com/rust-lang/cargo/issues/1479), 6~12 weeks), remove it:

> Thanks for this! This is a known issue, and we've actually got quite a small set of features to remove before Cargo builds on stable rust (as you've discovered!).
>
> At this time this unfortunately isn't actionable as all these features are crucial to the operation of Cargo and can't be removed. I plan to write an RFC for all the fs-related features quite soon, and git2's feature of catch_panic will likely also be stabilized during this next cycle (it was a late addition in the last one).

It all depends on when you want to merge this: defaulting it to on will make the build fail, ergo this will not be mergeable until cargo and its deps are buildable on the beta channel. Removal is possible, but I of course would rather have this merged sooner, therefore having it off and build from the nightly snapshot.

`"with-brewed-llvm"`: As for LLVM, I added it because rust vendors an LLVM that they patch with a few special-case performance patches, but it lags behind upstream which has received fixes to reported bugs, thus in some cases it is preferable to let someone build with a non-vendored LLVM. It's useful but I understand the will for removal.

@mikemcquaid, please confirm your choice for each option, and I shall proceed.